File: baekjoon/javascript/18258.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on("line", line => {
  input.push(line);
}).on("close", () => {
  input.shift();

  const Q = [];
  const result = [];
  for (let data of input) {
    let command = data.split(" ");
    if (command[0] == "push") Q.push(command[1]);
    else if (command[0] == "pop") result.push(Q.length ? Q.shift() : -1);
    else if (command[0] == "size") result.push(Q.length);
    else if (command[0] == "empty") result.push(Q.length ? 0 : 1);
    else if (command[0] == "front") result.push(Q.length ? Q[0] : -1);
    else if (command[0] == "back") result.push(Q.length ? Q[Q.length - 1] : -1);
    console.log(Q);
  }
  console.log(result.join("\n"));
});

File: baekjoon/python/18258.py
import sys; input = sys.stdin.readline
from collections import deque

q = deque()
result = []

for _ in range(int(input())):
    command = input().split()
    
    if command[0] == "push": 
        q.append(command[1])
    elif command[0] == "pop": 
        print(q.popleft() if q else -1)
    elif command[0] == "size": 
        print(len(q))
    elif command[0] == "empty": 
        print(0 if q else 1) 
    elif command[0] == "front":
        print(q[0] if q else -1)
    elif command[0] == "back": 
        print(q[-1] if q else -1)

# print("\n".join(map(str, result)))